src/reconcile/rec_visibility.c

```diff
@@ -404,12 +404,22 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
              * window ends when this tombstone started.
              */
             WT_ERR(__rec_append_orig_value(session, page, tombstone, vpack));
```

Comment:
@quchenhao 
Just to clarify my understanding:

Before this, we expect `last_upd->next == NULL`. And then after calling `__rec_append_orig_value`, the `next` pointer may get populated with an update representing the on-page value.

The `next` pointer might not get written to if the tombstone is currently globally visible since there's no reason to append an update behind a globally visible one. That is the scenario we're checking now.

Am I following?

Reply:
Yes, exactly.

Reply:
I'll add another assert to make sure last_upd->next is NULL before.